### PR TITLE
Refactor: refactoring the oauth2 login services

### PR DESCRIPTION
**Pull request by eapg**
* Refactoring the oauth2 login service
* Refactoring oauth2 login service test

### CODE REVIEW

File: lib/services/oauth2_service.ex [hunk 1]
```diff
@@ -50,6 +55,58 @@ defmodule EaRestaurantDataLoader.Lib.Services.Oauth2Service do
...
+    validate_client_credentials(client, client_secret)
+
+    [client_scope | _] = client.scopes
+
+    user = get_user_by_username_and_entity_status(username, Status.active())
+
+    validate_user_credentials(user, password)
+
+    access_token =
+      Oauth2Util.build_token(%{grant_type: "PASSWORD"}, %{
```
Comment (reynoldmorel) [inline lib/services/oauth2_service.ex]: Just send the `grant_type` coming from `login(...)`

File: lib/services/oauth2_service.ex [hunk 2]
```diff
@@ -21,20 +26,20 @@ defmodule EaRestaurantDataLoader.Lib.Services.Oauth2Service do
     [client_scope | _] = client.scopes
 
     access_token =
-      Oauth2Util.build_client_credentials_token(
-        client.client_name,
-        client_scope.scope,
-        client.access_token_expiration_time,
-        secret_key
-      )
+      Oauth2Util.build_token(%{grant_type: "CLIENT_CREDENTIALS"}, %{
```
Comment (reynoldmorel) [inline lib/services/oauth2_service.ex]: Same as below

File: lib/services/oauth2_service.ex [hunk 3]
```diff
@@ -98,6 +155,30 @@ defmodule EaRestaurantDataLoader.Lib.Services.Oauth2Service do
...
+          user
+
+        _ ->
+          raise InvalidCredentialsError
+      end
+
+    password_struct =
+      ApplicationUtil.build_password_type_from_env(password, validated_user.password)
+
+    IO.inspect(password_struct)
```
Comment (reynoldmorel) [inline lib/services/oauth2_service.ex]: I think this is wrong

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: lib/services/oauth2_service.ex (hunk 3 above)
Comment (eapg) [reply lib/services/oauth2_service.ex]: Done.

File: lib/services/oauth2_service.ex (hunk 1 above)
Comment (eapg) [reply lib/services/oauth2_service.ex]: Done.

File: lib/services/oauth2_service.ex (hunk 2 above)
Comment (eapg) [reply lib/services/oauth2_service.ex]: Done.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**